lib/enum/arrays.rb

```diff
@@ -1,24 +1,33 @@
 class Array
 
   def sum_of_numeric_elements
+    (self.group_by(&:class)[Fixnum] + self.group_by(&:class)[Float]).inject(:+)
```

Comment:
Your solution works but could be a lot simpler. We just need to do two things: first, select the elements that are both Fixnums and Floats, and then sum them. Conveniently, there is a class Numeric that fits both Fixnums and Floats. We can use select, like this: `self.select{ |item| item.is_a? Numeric }` and then use inject like you had it. 

Putting it all together: `self.select{ |item| item.is_a? Numeric }`.inject(:+)`